json5/model.py

```diff
@@ -46,6 +46,28 @@ def __init__(self, *key_value_pairs, trailing_comma=None, leading_wsc=None, tok=
         assert leading_wsc is None or all(isinstance(item, str) or isinstance(item, Comment) for item in leading_wsc)
         super().__init__(key_value_pairs=key_value_pairs, trailing_comma=trailing_comma, leading_wsc=leading_wsc or [], tok=tok)
 
+    def find_key_value_pair(self, key):
+        for key_value_pair in self.key_value_pairs:
+            compare_key =  key_value_pair.key
+            if isinstance(compare_key, String):
+                if compare_key.characters == key:
+                    return key_value_pair
+            elif isinstance(compare_key, Identifier):
+                if compare_key.name == key:
+                    return key_value_pair
+            else:
+                raise NotImplementedError(f"node type not handled as object key: {compare_key}")
```

Comment:
👍 I don't think there are other kinds of valid keys, but good that we guard this here just in case.